### PR TITLE
feat: Implement _mm_cvt_ps2pi

### PR DESCRIPTION
**Pull request by marktwtn**

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -4296,6 +4296,38 @@ FORCE_INLINE __m128 _mm_cvt_pi2ps(__m128 a, __m64 b)
...
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_cvt_ps2pi
+FORCE_INLINE __m64 _mm_cvt_ps2pi(__m128 a)
+{
+#if defined(__aarch64__)
+    return vreinterpret_m64_s32(
+        vget_low_s32(vcvtnq_s32_f32(vreinterpretq_f32_m128(a))));
+#else
+    int32_t res[2];
+    float32_t *data = (float32_t *) &a;
+    for (int i = 0; i < 2; i++) {
```
Comment (HowJMay) [inline sse2neon.h:4316]: Is it better to unfold it?

Comment (marktwtn) [reply sse2neon.h:4316]: For performance, it is.
I'll unfold it.

File: sse2neon.h [hunk 2]
```diff
@@ -4296,6 +4296,45 @@ FORCE_INLINE __m128 _mm_cvt_pi2ps(__m128 a, __m64 b)
...
+//   ENDFOR
+//
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_cvt_ps2pi
+FORCE_INLINE __m64 _mm_cvt_ps2pi(__m128 a)
+{
+#if defined(__aarch64__)
+    return vreinterpret_m64_s32(
+        vget_low_s32(vcvtnq_s32_f32(vreinterpretq_f32_m128(a))));
+#else
+    int32_t res[2];
```
Comment (jserv) [inline sse2neon.h:4314]: How about moving the following A32 specific implementation to `_mm_round_ps`? Thus, we can reimplement as following:
```c
a = _mm_round_ps(a, _MM_FROUND_CUR_DIRECTION);
return vcvt_s32_f32(vget_low_f32(a));
```

Comment (marktwtn) [reply sse2neon.h]: Done.

**Pull request merged**